types/block.go

```diff
@@ -1682,16 +1779,17 @@ func (blockID *BlockID) ToProto() tmproto.BlockID {
 	}
 
 	return tmproto.BlockID{
-		Hash:          blockID.Hash,
-		PartSetHeader: blockID.PartSetHeader.ToProto(),
+		Hash:                   blockID.Hash,
+		PartSetHeader:          blockID.PartSetHeader.ToProto(),
+		DataAvailabilityHeader: blockID.DataAvailabilityHeader.ToProto(),
 	}
 }
 
 // FromProto sets a protobuf BlockID to the given pointer.
 // It returns an error if the block id is invalid.
 func BlockIDFromProto(bID *tmproto.BlockID) (*BlockID, error) {
 	if bID == nil {
-		return nil, errors.New("nil BlockID")
+		return nil, nil
```

Comment:
is this change really necessary?

Reply:
at one point while debugging, it was, but now it's not. Good catch, thanks! https://github.com/lazyledger/lazyledger-core/pull/312/commits/a8a81ee7dad4fc7ef89c4d09d35e3fd24cbdb00a